internal/manifest/version.go

```diff
@@ -289,15 +297,26 @@ func (v *Version) Overlaps(
 					end = largest
 					restart = true
 				}
-				if restart {
-					ret = ret[:0]
-					continue loop
+			}
+
+			if restart {
+				// Need to retry the files we have not selected.
+				restart = false
```

Comment:
See other comment above that avoids the need for the `restart = false` statement.

A personal preference of mine (which you'll see frequently in the Pebble code base) is to avoid else-expressions when possible. Go-style also tends to encourage this. In this case, you could do:

```
if restart {
  continue
}

...
```

Reply:
i changed to 
if !restart {
}
let me know if you prefer something else